geom/src/tessellation.rs

```diff
@@ -19,16 +22,38 @@ pub struct Triangle {
 }
 
 impl From<Polygon> for Tessellation {
-    fn from(polygon: Polygon) -> Self {
+    fn from(mut polygon: Polygon) -> Self {
```

Comment:
I wonder if it'd be better for this be a borrowed Polygon. I see several `clones` elsewhere in this PR like:

```
Tessellation::from(self.blob.clone())
```

It seems like only the pre-tessellated polygons ever take advantage of owning the input, and for those cases, maybe it'd be better to have a consuming `take_tessellation` or `into_tessellation` method on Polygon.

No biggie, just a perf note.

Reply:
The normal code path winds up making a copy of every ring's points too... This seems unfortunate, since ultimately earcutr just [needs a &Vec<f64>](https://github.com/frewsxcv/earcutr/blob/2b865c29be08f153f06fb785388af8aca8e6e670/src/lib.rs#L750). Currently we're creating this intermediate nested GeoJSON form, then calling `flatten`. Probably copying the points could be avoided entirely with a little bit of cleverness, and maybe changing `earcutr` to take points as some kind of iterator.

Anyway, thanks for raising this. I'll make a note to get a profiling setup working and play around with variations of this idea!